src/cephadm/cephadm

```diff
@@ -5212,6 +5260,513 @@ def command_gather_facts():
+                s_time = time.time()
+
+                # list daemons should ideally be invoked with a fsid
+                data = list_daemons()
+                if not isinstance(data, list):
+                    errors.append("list-daemons didn't supply a list?")
+                    logger.warning(errors[-1])
+                    data = []
+                elapsed = time.time() - s_time
+                self.cephadm_cache.update_task(
+                    'daemons',
+                    {
+                        "scrape_timestamp": s_time,
+                        "scrape_duration_secs": elapsed,
+                        "scrape_errors": errors,
+                        "data": data,
+                    }
+                )
+                logger.debug(f"completed list-daemons scrape - {elapsed}s")
+            
+            time.sleep(CephadmDaemon.loop_delay)
+            ctr += CephadmDaemon.loop_delay
+        logger.info("list-daemons thread stopped")
+
+    def _create_thread(self, target, name, refresh_interval=None):
+        if refresh_interval:
+            t = Thread(target=target, args=(refresh_interval,))
+        else:
+            t = Thread(target=target)
+        t.daemon = True
+        t.name = name
+        self.cephadm_cache.update_health(name, "active")
+        t.start()
+
+        start_msg = f"Started {name} thread"
+        if refresh_interval:
+            logger.info(f"{start_msg}, with a refresh interval of {refresh_interval}s")
+        else:
+            logger.info(f"{start_msg}")
+        return t
+
+    def reload(self, *args):
+        """reload -HUP received
+        
+        This is a placeholder function only, and serves to provide the hook that could 
+        be exploited later if the exporter evolves to incorporate a config file
+        """
+        logger.info("Reload request received - ignoring, no action needed")
+
+    def shutdown(self, *args):
+        logger.info("Shutdown request received")
+        self.stop = True
+        self.http_server.shutdown()
+
+    def run(self):
+        logger.info(f"cephadm exporter starting for FSID '{self.fsid}'")
+        if not self.can_run:
+            logger.error("Unable to start")
+            return
+
+        # register signal handlers for running under systemd control
+        signal.signal(signal.SIGTERM, self.shutdown)
+        signal.signal(signal.SIGINT, self.shutdown)
+        signal.signal(signal.SIGHUP, self.reload)
+        logger.debug("Signal handlers attached")
+
+        host_facts = self._create_thread(self._scrape_host_facts, 'host', 5)
+        self.workers.append(host_facts)
+        
+        daemons = self._create_thread(self._scrape_list_daemons, 'daemons', 20)
+        self.workers.append(daemons)
+
+        disks = self._create_thread(self._scrape_ceph_volume, 'disks', 20)
+        self.workers.append(disks)
+
+        self.http_server = CephadmHTTPServer(('0.0.0.0', self.port), CephadmDaemonHandler)  # IPv4 only
```

Comment:
This means listening to all interfaces for well-known port...

Reply:
the port defaults to 5003 - but it's passed in at deployment time, it can be what you want.

Reply:
It was rather about limiting the service to the 'provisioning network' (if such a thing exists)...